### PR TITLE
Self formatting with 2.5.0-RC1

### PR DESCRIPTION
**Pull request by poslegm**
2.5.0-RC1 is available on Maven Central.

@kitbellew I see a lot of changes about `stripMargin` align. Is it intended by default? Diff in [CliArgParser](https://github.com/scalameta/scalafmt/compare/master...poslegm:self-formatting-2.5.0-rc1?expand=1#diff-90fa12de3f330db9b5e3bc1b91ec6055) looks weird a little.
Screen Shot 2020-04-15 at 6 40 42 PM

### CODE REVIEW

**Comment by poslegm** [comment]
Next steps if all changes is OK:

1. Write release notes
2. Publish it on GitHub releases
3. Attach native images 
[scalafmt-macos.zip](https://github.com/scalameta/scalafmt/files/4482196/scalafmt-macos.zip)
[scalafmt-linux.zip](https://github.com/scalameta/scalafmt/files/4482191/scalafmt-linux.zip)

**Comment by kitbellew** [comment]
> 2.5.0-RC1 is available on Maven Central.
> 
> @kitbellew I see a lot of changes about `stripMargin` align. Is it intended by default? Diff in [CliArgParser](https://github.com/scalameta/scalafmt/compare/master...poslegm:self-formatting-2.5.0-rc1?expand=1#diff-90fa12de3f330db9b5e3bc1b91ec6055) looks weird a little.
> Screen Shot 2020-04-15 at 6 40 42 PM

for `align=none`, `align.stripMargin=false`. for others, it would be a bug. if so, i need to fix it.

**Comment by kitbellew** [comment]
> Next steps if all changes is OK:
> 
> 1. Write release notes
> 2. Publish it on GitHub releases
> 3. Attach native images
>    [scalafmt-macos.zip](https://github.com/scalameta/scalafmt/files/4482196/scalafmt-macos.zip)
>    [scalafmt-linux.zip](https://github.com/scalameta/scalafmt/files/4482191/scalafmt-linux.zip)

I was planning to do a normal release, with notes etc., only on v.2.5.0. last time, the notes were spread between -rc1, -rc2 and release, which wasn't ideal, i thought.

**Comment by poslegm** [comment]
> for align=none, align.stripMargin=false.

Ok, then I wrote about it in migration guide.

> the notes were spread between -rc1, -rc2 and release, which wasn't ideal, i thought.

Autogenerated release notes with native binaries are necessary for every release whatever. I'll take care of it.

File: scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtConfig.scala [hunk 1]
```diff
@@ -327,13 +328,14 @@ object ScalafmtConfig {
     )
   }.map { case (k, v) => k.toLowerCase -> v }
 
-  def conservativeRunner: ScalafmtRunner = default.runner.copy(
-    optimizer = default.runner.optimizer.copy(
-      // The tests were not written in this style
-      forceConfigStyleOnOffset = 500,
-      forceConfigStyleMinArgCount = 5
+  def conservativeRunner: ScalafmtRunner =
```
Comment (kitbellew) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtConfig.scala:331]: while i am ok with this, it's the result of the fix to `alwaysBeforeMultilineDef` (which wasn't working properly before). if you wish, we can disable the flag, see if that reduces diff.

**Comment by kitbellew** [comment]
> > for align=none, align.stripMargin=false.
> 
> Ok, then I wrote about it in migration guide.

well, there's the bug. apparently, @olafurpg was right and you can't use `align=none` and `align.xxx = yyy`. let me see if i can fix that.

File: scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtConfig.scala (hunk 1 above)
Comment (poslegm) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtConfig.scala:331]: No, I am ok with it

**Comment by poslegm** [comment]
May be just move `align.stripMargin` from `align` block? Strings are literals also. So we can introduce
```
literals.assumeStandardLibraryStripMargin (and deprecate current)
literals.alignStripMargin
```

**Pull request merged**